### ISSUE TITLE
The destination wallet didnot receive IBC token

### ISSUE THREAD

**Issue by cifer-ai** [informalsystems/hermes#3941]
## Summary of Bug

As last time IBC token transferring seems fail because we have change our chain-id.

This time we create new chain, with chain id "cifer-2" and create new client, connection, channel between cifer-2 and osmosis-1
There's a lot of connection error but we try multiple times until it success.

IBC channel-39205 (osmosis-1 to cifer-2)
https://chainsco.pe/osmosis/ibc/channels/channel-39205%2Ftransfer

Then we try sending ibc token (cif to osmo wallet) using this command

`hermes tx ft-transfer --timeout-seconds 1000 --dst-chain osmosis-1 --src-chain cifer-2 --src-port transfer --src-channel channel-1 --amount 1 --denom ucif`

We get this output after doing token transfer
```
SUCCESS [
    IbcEventWithHeight {
        event: SendPacket(
            SendPacket {
                packet: Packet {
                    sequence: 2,
                    source_port: PortId(
                        "transfer",
                    ),
                    source_channel: ChannelId(
                        "channel-1",
                    ),
                    destination_port: PortId(
                        "transfer",
                    ),
                    destination_channel: ChannelId(
                        "channel-39205",
                    ),
                    data: [123, 34, 97, 109, 111, 117, 110, 116, 34, 58, 34, 49, 34, 44, 34, 100, 101, 110, 111, 109, 34, 58, 34, 117, 99, 105, 102, 34, 44, 34, 114, 101, 99, 101, 105, 118, 101, 114, 34, 58, 34, 111, 115, 109, 111, 49, 108, 122, 117, 120, 50, 108, 119, 117, 56, 120, 55, 99, 103, 107, 52, 113, 119, 56, 114, 54, 104, 54, 106, 120, 103, 122, 110, 115, 52, 54, 53, 117, 99, 121, 101, 100, 53, 50, 34, 44, 34, 115, 101, 110, 100, 101, 114, 34, 58, 34, 99, 105, 102, 101, 49, 54, 104, 102, 122, 51, 97, 118, 106, 48, 56, 112, 110, 119, 121, 56, 48, 122, 54, 122, 99, 103, 114, 122, 108, 116, 104, 115, 99, 57, 106, 50, 48, 116, 102, 57, 107, 99, 113, 34, 125],
                    timeout_height: Never,
                    timeout_timestamp: Timestamp {
                        time: Some(
                            Time(
                                2024-04-10 3:45:37.836364452,
                            ),
                        ),
                    },
                },
            },
        ),
        height: Height {
            revision: 2,
            height: 1374,
        },
    },
]
```
The problem remain the same, success outbound (the token in cifer wallet decreased) but the inbound was unsuccessful (there's no ibc token received in destination wallet)

We use this command to query bank balance.
`osmosisd query bank balances [osmo_walletaddress]`

When we try sending osmo to cifer wallet
`hermes tx ft-transfer --timeout-seconds 1000 --dst-chain cifer-2 --src-chain osmosis-1 --src-port transfer --src-channel channel-39205 --amount 1 --denom uosmo`

We got this error
```
ERROR gRPC call `query_connection` failed with status: status: Unavailable, message: "grpc-status header missing, mapped from HTTP status code 429", details: [], metadata: MetadataMap { headers: {"date": "Wed, 10 Apr 2024 03:29:55 GMT", "content-length": "0", "ratelimit-policy": "10; w=60", "ratelimit-policy": "10; w=60", "ratelimit-policy": "10; w=60", "ratelimit-policy": "10; w=60", "ratelimit-policy": "1; w=1", "ratelimit-policy": "1; w=1", "ratelimit-policy": "1; w=1", "ratelimit-policy": "1; w=1", "cf-cache-status": "DYNAMIC", "report-to": "{\"endpoints\":[{\"url\":\"https:\/\/a.nel.cloudflare.com\/report\/v4?s=%2F85TgG0x7enkNdXTUtpWIXuDbeAAGyrF6hMscj5FgCXgEZplQVziB8iaXOhqwglpxE6%2BYN%2BuAWwBPg5x8SwmwyvPyDTLmyX%2F%2FnqaaUMESHiMBtFqDvjWP7Fa4AL65ooiGH4oIxMSco5EZCHJ%2FAUh\"}],\"group\":\"cf-nel\",\"max_age\":604800}", "nel": "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}", "server": "cloudflare", "cf-ray": "871fa7c0ce73a18d-BKK"} }
```

---

When we try sending atom from cosmos wallet to osmosis wallet, it was successfully transfer.
There is IBC token for atom in destination wallet.

The transferred IBC atom token appear in this wallet, but not cif token.
```
balances:
- amount: "20"
  denom: ibc/10E9601BA1D5C24E165A3E761F59384CAF634FE473011962CE28E65AAFDA30F0
- amount: "8208544"
  denom: uosmo
pagination:
  next_key: null
  total: "0"
```

---

We check Hermes config using `hermes health-check` and it turns out all chains are healthy.

You may check our node status here:
https://cosmos.directory/cifer

## Version

hermes 1.8.2+06dfbaf

## Acceptance Criteria

Please give us steps to troubleshoot these errors.

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by romac** [comment]
> The problem remain the same, success outbound (the token in cifer wallet decreased) but the inbound was unsuccessful (there's no ibc token received in destination wallet)

Are you running `hermes start` in the background when you do the `tx ft-transfer` command? Otherwise, nothing will be relayed, `ft-transfer` only submits the Transfer msg to the chain but that's it. You need Hermes running alongside in order to relay the acks etc. If things still don't work with Hermes running, please send the relevant logs from `hermes start` at `debug` or `trace` level.

> We got this error

> ERROR gRPC call `query_connection` failed with status: status: Unavailable, message: "grpc-status header missing, mapped from HTTP status code 429", details: [], metadata: MetadataMap { headers: {"date": "Wed, 10 Apr 2024 03:29:55 GMT", "content-length": "0", "ratelimit-policy": "10; w=60", "ratelimit-policy": "10; w=60", "ratelimit-policy": "10; w=60", "ratelimit-policy": "10; w=60", "ratelimit-policy": "1; w=1", "ratelimit-policy": "1; w=1", "ratelimit-policy": "1; w=1", "ratelimit-policy": "1; w=1", "cf-cache-status": "DYNAMIC", "report-to": "{\"endpoints\":[{\"url\":\"https:\/\/a.nel.cloudflare.com\/report\/v4?s=%2F85TgG0x7enkNdXTUtpWIXuDbeAAGyrF6hMscj5FgCXgEZplQVziB8iaXOhqwglpxE6%2BYN%2BuAWwBPg5x8SwmwyvPyDTLmyX%2F%2FnqaaUMESHiMBtFqDvjWP7Fa4AL65ooiGH4oIxMSco5EZCHJ%2FAUh\"}],\"group\":\"cf-nel\",\"max_age\":604800}", "nel": "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}", "server": "cloudflare", "cf-ray": "871fa7c0ce73a18d-BKK"} }

You are getting rate limited by Cloudflare, nothing we can do about that. Hermes is designed to work best when running on bare metal, together with the full nodes of all the chains it needs to relay between.

**Comment by cifer-ai** [comment]
Actually we ran `hermes start` in the background all the time when we try transferring, but still didn't work.

Anyway, after a while, we do `hermes health-check` again, turn out Osmosis chain is not healthy.

In conclusion, we change rpc endpoints of Osmosis multiple times until it's healthy and transfer token.
Then using `hermes tx packet-recv` for multiple times, until tokens appear in destination wallet.

So it seems to be connection problem, and some Cloudflare rate limited of some nodes on Osmosis side.

**Comment by cifer-ai** [comment]
Actually we ran `hermes start` in the background all the time when we try transferring.

On our side, we set longer bonding period and try again, but it still didn't work.

Anyway, after a while, we do `hermes health-check` again, turn out Osmosis chain is not healthy.

In conclusion, we change rpc endpoints of Osmosis multiple times until it's healthy and transfer token.
Then using `hermes tx packet-recv` for multiple times, until tokens appear in destination wallet.

So it seems to be connection problem, and some Cloudflare rate limited of some nodes on Osmosis side.
